src/isolate/server/_server.py

```diff
@@ -28,9 +31,45 @@
 RUN_STORE: Dict[str, RunInfo] = {}
 MAX_JOIN_WAIT = 1
 
+app.config.from_prefixed_env("FAL")
+
+if app.config.get('USER_NAME', False):
+    app.config['SECRET_KEY'] = secrets.token_urlsafe()
+    app.config['AUTH_TOKEN'] = create_auth_token(
+        app.config['USER_NAME'],
+        app.config['SECRET_KEY'])
+
+    print("++++")
+    print("The following line contains your authentication token. Put it in 'x-access-token' header.")
+    print(app.config['AUTH_TOKEN'])
+    print("++++")
+
+# Authentication decorator
+def check_auth(f):
+    @wraps(f)
+    def decorator(*args, **kwargs):
+        token = None
+        if 'x-access-token' in request.headers:
+            token = request.headers['x-access-token']
+            if validate_auth_token(
+                    token,
+                    app.config.get("USER_NAME", False),
+                    app.config.get("SECRET_KEY", False)):
```

Comment:
And then, I would guess no defaults needed
```suggestion
                    app.config["USER_NAME"],
                    app.config["SECRET_KEY"]):
```